src/Symfony/Component/ExpressionLanguage/Compiler.php

```diff
@@ -21,9 +21,9 @@ class Compiler
     private $source;
     private $functions;
 
-    public function __construct(array $functions)
+    public function __construct(array &$functions)
```

Comment:
I am not sure if passing the functions by reference ID actually a good idea as it may lead to unexpected side effects.